src/prefect/server/utilities/subscriptions.py

```diff
@@ -21,16 +28,49 @@ async def accept_prefect_socket(websocket: WebSocket) -> Optional[WebSocket]:
         # message is expected to be an auth message, and if any other type of
         # message is received then the connection will be closed.
         #
-        # There is no authentication in Prefect Server, but the protocol requires
-        # that we receive and return the auth message for compatibility with Prefect
-        # Cloud.
+        # The protocol requires receiving an auth message for compatibility
+        # with Prefect Cloud, even if server-side auth is not configured.
         message = await websocket.receive_json()
-        if message["type"] != "auth":
+
+        auth_setting = PREFECT_SERVER_API_AUTH_STRING.value()
```

Comment:
Can you use `get_current_settings` here? I think we should eventually move away from accessing settings this way since it's less type safe.